sunpy/map/mapbase.py

```diff
@@ -1638,16 +1646,22 @@ def plot(self, annotate=True, axes=None, title=True, **imshow_kwargs):
                 axes.set_title(title)
 
             # x-axis label
-            if self.coordinate_system.x == 'HG':
-                xlabel = 'Longitude [{lon}]'.format(lon=self.spatial_units.x)
+            if self.coordinate_system.lon.startswith('HGLN'):
+                xlabel = 'Heliographic Longitude [{lon}]'.format(lon=self.spatial_units.lon)
+            elif self.coordinate_system.lon.startswith('CRLN'):
+                xlabel = 'Carrington Longitude [{lon}]'.format(lon=self.spatial_units.lon)
+            elif self.coordinate_system.lon.startswith('HPLN'):
+                xlabel = 'Helioprojective Longitude (Solar-X) [{xpos}]'.format(xpos=self.spatial_units.lon)
```

Comment:
Finally!! not a 'X-position' but a proper information on the coordinates!!! :tada: